# Why people wash meat (or don't)

YouTuber (and former public radio reporter) [Adam Ragusea](https://www.youtube.com/user/aragusea/about) [asked his viewers](https://youtu.be/U_PMnCpaJiQ?t=742) in September 2020 to answer a detailed survey about whether (and why and how) they wash meat before cooking it. He received more than 13,000 responses. The data explored and analyzed here is the anomymized data collected. Ragusea published a [video](https://www.youtube.com/watch?v=90Nd_vh3yk8) explaining more around washing meat and his analysis of the data.

[USDA Study: Food Safety Consumer Research Project: Meal Preparation Experiment Related to Poultry Washing](https://www.fsis.usda.gov/wps/wcm/connect/9bb3a252-e12e-40e5-b76b-cb46a2322c3f/FSCRP_Year+2_Final_Aug2019.pdf?MOD=AJPERES) - Published: August 2019

["Don't Wash Your Chicken!" Results of an Interdiciplinary Approach to Reduce Incidence of Infectious Foodborne Diseases](https://idea.library.drexel.edu/islandora/object/idea%3A4369/datastream/OBJ/view) - December 2013

<i>Due to the nature of how this data was gathered this was NOT a random sample. Therefore results, analyses, conclusions, and insights should be treated as such. </i>


In [2]:
import numpy as np
import pandas as pd


In [3]:
df_survey = pd.read_csv('data/RaguseaMeatWashingSurveyResponsesRaw.csv')

df_survey.sample(10)

/Users/bryandunn/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3146: DtypeWarning: Columns (2,10,11,27,28,29,30,31,32,34,35,36,37,38,39,40,52,53,54,55,56,57,63,64,67) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Timestamp Do you prepare meat for yourself or others ?  \
5112   9/28/20 17:59                                    Sometimes   
5413   9/28/20 18:21                                        Often   
12107  9/30/20 17:13                                        Often   
11973  9/30/20 14:40                                    Sometimes   
12068  9/30/20 16:29                                    Sometimes   
4289   9/28/20 17:09                                    Sometimes   
4656   9/28/20 17:29                                        Often   
7751   9/28/20 23:01                                        Often   
6837   9/28/20 20:47                                    Sometimes   
10977  9/29/20 18:19                                        Often   

      What is your current age?   \
5112                          13   
5413                          32   
12107                         23   
11973                         29   
12068                         28   
4289                          17   
4656                          21   
7751                          33   
6837                          24   
10977                         21   

      What is your gender ? (Self-describe using 'other.')  \
5112                                                Male     
5413                                                Male     
12107                                               Male     
11973                                               Male     
12068                                               Male     
4289                                                Male     
4656                                                Male     
7751                                              Female     
6837                                                Male     
10977                                               Male     

            Identify your country or territory of origin  \
5112                            United States of America   
5413                                              Poland   
12107  United Kingdom of Great Britain and Northern I...   
11973                                            Germany   
12068                                           Slovakia   
4289                            United States of America   
4656                                               Spain   
7751                            United States of America   
6837                            United States of America   
10977                                           Bulgaria   

      Identify your current country or territory of residency  \
5112                            United States of America        
5413                                              Poland        
12107  United Kingdom of Great Britain and Northern I...        
11973                                            Germany        
12068                                     Czech Republic        
4289                            United States of America        
4656                                               Spain        
7751                            United States of America        
6837                            United States of America        
10977                                           Bulgaria        

      Current U.S. state of residence   \
5112                          Oklahoma   
5413                               NaN   
12107                              NaN   
11973                              NaN   
12068                              NaN   
4289                           Arizona   
4656                               NaN   
7751                          Illinois   
6837                           Florida   
10977                              NaN   

      What is your "race"? (Categories are those used by the U.S. Census Bureau — check all that apply.)  \
5112                    American Indian or Alaska Native                                                   
5413                                                 NaN                                                   
12107  

In [4]:
df_survey.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13528 entries, 0 to 13527
Data columns (total 68 columns):
 #   Column                                                                                                                                                                                                                                                                                                                                                                                                                  Non-Null Count  Dtype 
---  ------                                                                                                                                                                                                                                                                                                                                                                                                                  --------------  ----- 
 0   Timestamp     

In [7]:
df_survey['For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Pork (pigs)]'].value_counts()


Dont Wash     10414
Wash           1443
Wash Pork?        1
Name: For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Pork (pigs)], dtype: int64

<div class="alert alert-block alert-info">


### SUMMARY

There are 1352 raw rows of data in the dataframe, meaning there were 13,528 respondants to the survey about washing meat. There where 67 questions contained in the survey. 


`df_survey` columns:

NOTES: These are the RENAMED column titles. 

* <b>
    </div>

## Cleaning Plan Summary

##### Tidiness Issues:
1. Column names often contain full sentences of the questions from the survey. 

##### Quality Issues:

1. Missing 

In [ ]:
## Clean Data


<div class="alert alert-block alert-warning">

<b>Tidiness Issue #1:</b> Column names often contain full sentences of the questions from the survey. 

SOLUTION: Column names are renamed to more concise title. 
    </div>

In [ ]:
# rename 'tweet_id' to 'id' in twitter_archive
df_survey.rename(columns={'Do you prepare meat for yourself or others ?' : 'prepare_meat', 
                          'What is your current age?' : 'age', 
                          "What is your gender ? (Self-describe using 'other.')" : 'gender', 
                          'Identify your country or territory of origin' : 'country_origin',
                          'Identify your current country or territory of residency' : 'country_residency',
                          'Current U.S. state of residence' :
                          'What is your "race"? (Categories are those used by the U.S. Census Bureau — check all that apply.)' :
                          'Are you of Hispanic, Latino, or Spanish origin?' :
                          'How often do you prepare a meal in your household?' :
                          'How many people are in your household?' :
                          'How  important is food safety to you when you cook?' :
                          'Which of the following do you do regularly when cooking, preparing and consuming food products? (Check all that apply)' :
                          'Before you begin to prepare food in your kitchen, how often do you wash your hands with soap and warm water for at least 20 seconds?' :
                          'Please indicate how often you do each of the following [Use different plates for handling raw meat and cooked meat]' : 
                          'Please indicate how often you do each of the following [Use a food thermometer to check the doneness for red meat, pork or poultry]' :
                          'Please indicate how often you do each of the following [Not wash utensils used to handle raw food before they are used for cooked food]' :
                          'Please indicate how often you do each of the following [Use the same cutting board without cleaning it with hot soapy water between using it for raw meat or chicken and using it for ready-to-eat food]' :
                          'Please indicate how often you do each of the following [Leave perishable food out of the refrigerator for more than two hours]' :
                          'For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Pork (pigs)]' :
                          'For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Poultry (chicken, turkey, duck, any other bird)]' :
                          'For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Beef (cattle or buffalo)]' :
                          'For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Sheep (lamb) or goat]' :
                          'For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Fish (marine, fresh water, finned, shellfish, etc.)]' :
                          'For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Ground meat]' :
                          'For each of the following raw meats, please check the box if you commonly wash or do not wash it before cooking. If you don’t eat it at all, leave the boxes blank. (Categories are those identified by the U.N. FAO as the most globally popular meats, plus fish and ground meat.) [Other]' :
                          'If you said "other," please explain.' :
                          'If you don’t commonly wash raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I feel like I should wash the meat, but I don’t.]' :
                          'If you don’t commonly wash raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I believe the meat is already clean enough.]' :
                          'If you don’t commonly wash raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I’ve heard some food safety experts recommend against washing meat.]' :
                          'If you don’t commonly wash raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I want to preserve the taste, smell, texture or appearance of the meat.]' :
                          'If you don’t commonly wash raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [It is not a cultural or religious custom or tradition for me.]' :
                          'If you don’t commonly wash raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [Other]' :
                          'If you said "other," please explain..1' : 
                          'If you do commonly wash any raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I’m concerned about the health and safety of the meat.]' :
                          'If you do commonly wash any raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I’m acting in accordance with a religious belief or practice.]' :
                          'If you do commonly wash any raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I’m trying to improve the taste or smell of the meat.]' :
                          'If you do commonly wash any raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I’m trying to improve the texture of the meat.]' :
                          'If you do commonly wash any raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [I’m trying to improve the appearance of the meat.]' :
                          'If you do commonly wash any raw meat before cooking, please rank the following reasons, with #1 being the most relevant to you, #2 next most relevant, and so on. If something isn’t relevant, leave the box blank. (You may only choose one row per ranking column.) [Other]' :
                          'If you said "other," please explain..2' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Pathogens (bacteria, viruses, fungi, etc.)]' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Artificial chemicals (preservatives, pesticides, cleaning solutions, etc.)]' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Inedible debris (dirt, dust, bone fragments, feathers, etc.)]' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Undesirable flavors or odors]' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Blood]' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Slime]' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Fat]' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Meat juice (the pink, non-blood liquid that comes out of muscle tissue)]' :
                          'If you do commonly wash any raw meat before cooking, what are you trying to remove? Rank the following concerns in order of their importance to you, with #1 being the most important, #2 the next most important, and so on. If something isn’t a concern to you, leave the box blank. (You may only choose one row per ranking column.) [Other]' :
                          'If you said "other," please explain..3' :
                          'If you do commonly wash any raw meat before cooking, please rank the methods you use, with #1 being the method you use most often, #2 next most often, and so on. If you don’t use the method at all, leave the box blank. (You may only choose one row per ranking column.) [Rinsing under running water]' :
                          'If you do commonly wash any raw meat before cooking, please rank the methods you use, with #1 being the method you use most often, #2 next most often, and so on. If you don’t use the method at all, leave the box blank. (You may only choose one row per ranking column.) [Rinsing in a bowl or basin of plain water]' :
                          'If you do commonly wash any raw meat before cooking, please rank the methods you use, with #1 being the method you use most often, #2 next most often, and so on. If you don’t use the method at all, leave the box blank. (You may only choose one row per ranking column.) [Rinsing in anything other than plain water (salt brine, vinegar, citrus, etc.)]' :
                          'If you do commonly wash any raw meat before cooking, please rank the methods you use, with #1 being the method you use most often, #2 next most often, and so on. If you don’t use the method at all, leave the box blank. (You may only choose one row per ranking column.) [Soaking awhile in plain water]' :
                          'If you do commonly wash any raw meat before cooking, please rank the methods you use, with #1 being the method you use most often, #2 next most often, and so on. If you don’t use the method at all, leave the box blank. (You may only choose one row per ranking column.) [Other]' :
                          'If you said "other," please explain..4' :
                          'If you commonly rinse or soak meat before cooking, you most-often use:' :
                          'If you commonly soak raw meat before cooking, please indicate how long.' :
                          'If you do commonly rinse or soak raw meat in anything other than plain water, which of these things do you use? Check as many as apply.' :
                          'If you do commonly wash any raw meat before cooking, how do you clean up afterward? Check any that apply to you.' :
                          "When you eat at someone's house, how important is it to you that they be attentive to food safety in their food preparation?" :
                          'When you eat at a commercial eatery, how important is it to you that they be attentive to food safety in their food preparation?':
                          'Is there anything else you wish to share about your food preparation practices?' :
                          'Are there comments about consumer food safety education or food guidelines that you wish to share?'
                          
                                     }, inplace=True)

# Visualizations

In [ ]:
# gender proportion - pie chart

In [ ]:
# top 5 countries

In [ ]:
# top 10 states (USA)

In [ ]:
# race distribution

In [ ]:
# age - histogram